pages/security/csp.md

```diff
@@ -0,0 +1,130 @@
+
+`<meta http-equiv="Content-Security-Policy" content="default-src 'self'; img-src https://*; child-src 'none';">`
+
+You're done!
+
+### Server-side Implementation
+To include a CSP headers on the server, simply return a `Content-Security-Policy: {my-policy-string}` HTTP header in your web responses.
+
+<br>
+
+Although a basic CSP policy is relatively easy to write, you might find it easier to use a server-side library to define your policies declaratively. Some well-known and actively maintained libraries include:
+
+* [secureheaders](https://github.com/twitter/secureheaders), a **Ruby** gem written and maintained by Twitter
+* [helmetjs](https://github.com/helmetjs/helmet), middleware for **Express.js**
+* [django-csp](https://github.com/mozilla/django-csp), maintained by Mozilla.
+
+Note that this list is neither exhaustive nor prescriptive. These libraries are included in this guide to give you a sense of the capabilities that might be included in a library that works with the CSP header.
+
+## Caveats
+A major caveat is that CSP, by default, prevents developers from adding inline `<style>` and `<script>` tags, and from specifying JavaScript behavior or CSS `style` attributes directly on DOM elements. This isn't a bad thing, as current best practices in front-end development discourage the use of these patterns.
+
+However, there are scenarios in which you may want to use an inline content tag, such as including a Google Analytics initialization script. For these cases, CSP allows the user to supply a `nonce` attribute to the inline script: `<script nonce=GHDGsfsd83hfdfFD3>...javascript here...</script>`. Then, in your policy, `script-src 'nonce-GHDGsfsd83hfdfFD3'`. Nonces must be regenerated on each request. Additionally, CSP 2.0 allows you to generate a SHA hash of the script content, and pass that to the `srcipt-src` directive as `'sha-{content-sha}'`. You don't need to use a `nonce` attribute if your content is hashed.
+
+Another potential issue is third-party libraries that automatically inject JavaScript and CSS into your HTML. If your project utilizes a library that does this, the only guaranteed solution is
+to use the `unsafe-inline` value when setting the `script-src` directive; this obviously defeats the purpose of having a CSP for your JavaScript.
+
+If you must load external scripts inline and are not allowed to use the `unsafe-inline` keyword, you could also make a SHA hash of the script being included, and whitelist that in your content security policy.
+
+This technique will allow those scripts to load, with the following caveats:
+
+* Each time the third-party script changes, a new hash will have to be computed, and your policy's `script-src` whitelist will need to be updated
+* Inlining the SHAs of multiple scripts adds bloat to the policy, and the number of bytes needed to transmit the header to the browser.
```

Comment:
There's a word missing in the second half. Should it be "and increases the number of bytes"?